libbeat/common/docker/util.go

```diff
@@ -0,0 +1,47 @@
+package docker
+
+import (
+	"net"
+	"os"
+	"strings"
+)
+
+// GetHostIP can look at the hostname and give the container gateway/host IP based on where the given
+// beat is running.
+func GetHostIP(watcher Watcher) string {
+	hostname, err := os.Hostname()
+	if watcher == nil || err != nil {
+		return ""
+	}
+
+	// Assume that hostname is the container ID and attempt to do docker inspect to get its Gateway address.
+	// Gateway address of a container is one of the interfaces on the host that can be used to connect to
+	// containers that are being run on host network.
+	for cid, container := range watcher.Containers() {
+		if strings.Index(cid, hostname) == 0 && container != nil {
+			if len(container.gateways) != 0 {
+				return container.gateways[0]
+			}
+		}
+	}
+
+	// If the process is not running in a container then return a valid IPv4 address that can be used to access
+	// the service as it would be running on host network.
+	return getLocalIP()
+}
+
+func getLocalIP() string {
```

Comment:
This method is a bit tricky, are we sure that `net.InterfaceAddrs()` always returns the addresses in the same order? what happens if an address is added or removed?

I think there is no safe method to guess the address of a host with multiple addresses :neutral_face: maybe for that case we should allow to configure a host address in metricbeat (e.g. sensu has the [address setting](https://docs.sensu.io/sensu-core/1.2/reference/clients/#client-attributes), and kubelet has the `--node-ip` flag), and take it from there @ruflin wdyt?